claat/proto-renderer/generic_renderer.go

```diff
@@ -0,0 +1,19 @@
+package genrenderer
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"text/template"
+)
+
+// ExecuteTemplate returns the evaluated template per passed templating
+// namespace, based on the passed Codelab proto element type
+func ExecuteTemplate(el interface{}, t *template.Template) string {
+	var w bytes.Buffer
+	e := t.ExecuteTemplate(&w, reflect.TypeOf(el).Name(), el)
```

Comment:
Per PR description, this will change to a function-based type-switch on a later PR. This is the base functionality per this setup PR.

Reply:
ideally this will be a proto, but the definitions of leaf-types need to be redefined first. Showcasing generality here.

Reply:
If this is expected to change eventually, is there a reason we're not just changing it now?

Reply:
- Proto definitions need to be redefined and softly-agreed on
- Smaller reviews